front-end/src/screens/Settings/fullname.tsx

```diff
@@ -3,19 +3,21 @@ import React, { useState, useContext, useEffect } from 'react'
 import Button from '../../ui-components/Button'
 import { Form } from '../../ui-components/Form'
 import { useChangeNameMutation } from '../../generated/graphql'
-import { UserDetailsContext } from '../../context/UserDetailsContext'
 import { NotificationContext } from '../../context/NotificationContext'
 import { NotificationStatus } from '../../types'
 
-const Fullname = (): JSX.Element => {
+interface Props {
+	profileName?: string | null | undefined
```

Comment:
Same here, why not from the context?

Reply:
The reason i am loading from user profile query is because context coming from JWT after update JWT becomes stale with old data. This causes a bug that o refresh you see old data.

Should i update JWT from refresh token on updates and get data from context?

Reply:
ah I see. The context should not be directly dependant from the JWT I believe. Actually it is not. It is initialized with the JWT but gets updated when we login/logout (from the server's answer in [auth-service.tsx](https://github.com/paritytech/polkassembly/blob/master/front-end/src/services/auth.service.tsx#L35), and then whenever there's a new value. The change will be reflected in the next JWT we'll get.)

I didn't do big tests but the JWT as well as the context should end up with those values. The context (and the server) have the same most up to date values.

What do you think?

Reply:
check now. Also how to refresh token after update?

Reply:
getting new token in change response